app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoFetchScheduler.kt

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pocket
+
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.PRIVATE
+import androidx.lifecycle.Lifecycle.Event.ON_START
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.work.Constraints
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import java.util.Calendar
+import java.util.concurrent.TimeUnit
+import kotlin.random.Random
+
+private const val FETCH_UNIQUE_WORK_NAME = "PocketFetch"
+
+/**
+ * Schedules background fetches of the Pocket video data.
+ */
+class PocketVideoFetchScheduler : LifecycleObserver {
+
+    @OnLifecycleEvent(ON_START)
+    fun onStart() {
+        schedulePocketBackgroundFetch(WorkManager.getInstance())
+    }
+
+    @VisibleForTesting(otherwise = PRIVATE)
+    fun schedulePocketBackgroundFetch(workManager: WorkManager) {
+        val constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+
+        // TODO: why do we want this from a UX perspective.
+        // todo: this explanation sucks.
+        // Our fetch scheduling scheme: when the user foregrounds the app, we schedule an update for the next calendar
+        // day (i.e. if today is the 2nd, we schedule for the 3rd) at a random time inside our fetch interval.
+        // it's still "tonight" but already after midnight, we
+        // attempt to schedule one for the following night.
```

Comment:
Does this mean that a daily user would never receive Pocket updates?

Reply:
What do you mean by a daily user? This comment is really bad and needs to be improved. :(

The only people who would not receive updates are people who leave the app open all the time without a screensaver (because onStart is never called again to schedule background fetching or push in new content).